### DISCUSSION TITLE
Is the waveshare WI-FI version suitable

### DISCUSSION THREAD

**Discussion by magichew** [alienatedsec/solis-ha-modbus-cloud#8] [Q&A] [answered]
Hi

I found this on Amazon (Waveshare RS485 to WiFi/Ethernet Module Rail-Mount Serial Server Modbus MQTT Gateway Supports Transparent Transmission Mode https://amzn.eu/d/3fveYKC)

![51sI4hVsA3L _AC_UF1000,1000_QL80_](https://user-images.githubusercontent.com/5846333/231965398-0ef6ffdf-ad21-4013-a1f0-11b61b9a01f8.jpg)

Would this work for those who are having a hard time getting network cable to and fro but have power handy?

**Comment by alienatedsec** [comment, marked as answer]
Hi @magichew 

While not tested by myself, I would not recommend it the same way as it is described on the SolaX Integration website.
https://github.com/wills106/homeassistant-solax-modbus/wiki/Compatible-RS485-Adaptors#waveshare-rs485-to-wifieth

I hope that helps.

However, may be worth investing in some Powerline Adaptors.
https://www.amazon.co.uk/s?k=powerline+adapter

**Reply by alienatedsec** [reply]
[Waveshare RS485 to WIFI/ETH](https://www.waveshare.com/wiki/RS485_TO_WIFI/ETH)
This RS485 adaptor is very problematic:

- There seems to be delayed communications,
- Struggles to read more than 7 registers at a time,
- Modbus Error: [Input/Output] Unable to decode request,
- No ability to set "Auto query storage type" for Sofar Inverters (also needed for the TCP Server)

**Comment by magichew** [comment]
I actually received two of the POE versions of the RS45 to ETH yesterday but had something shaking around inside of it, sure enough when I plugged it in it was dead so I need a replacement. Was going to try the WIFI version but I'll steer clear now.

**Reply by magichew** [reply]
Actually, just opened it up and what I presume is the POE module was loose. Fitted it back to the board and plugged it in and it has lit up. Now on to configuring.

**Reply by alienatedsec** [reply]
How is your setup going @magichew ?

**Reply by magichew** [reply]
Struggling to be honest. Waiting for the plug/socket to arrive so thought I'd set up the Waveshare units ahead of time. Got Parallels going and installed Vircom. I've tried running a search for the Waveshares but they're not showing up on the network.

Will they be assigned an IP address via DHCP or do I need to set up a local only connection directly via the computer?

**Reply by alienatedsec** [reply]
Once they are connected to Ethernet, they should be visible in the Vircom regardless of their IP. They will make some noise on the network, but you can configure them individually to use DHCP afterwards.

**Reply by magichew** [reply]
So if I load up Vircom, I click Device then Auto Search and nothing happens. I then click Add Manually and Set my IP range and then search begins but never finds anything.

Wonder if it is a Parallels thing, shouldn't be though.

**Reply by hazbot72** [reply]
what type of virtual network adaptor do you have set in Parallels for your VM?   If it is set to type "NAT" , it maybe can't see broadcast traffic on your home network. Try setting the adaptor to type "Bridged" or whatever is the equivalent in Parallels.

Sorry , I'm not familiar with Parallels specifically. But the above would apply for other virtualisation tools such as VirtualBox etc

**Reply by magichew** [reply]
Hi, thanks, I think I've altered that setting assumed the default was bridged network (incorrectly).

Now one device is showing in Vircom. Should both show? Or do I need to configure one at a time.

**Reply by hazbot72** [reply]
hmm yes in my case, both of the devices appeared in Vircom right out of the box.

Hopefully it's not the case that whichever of your devices had the loose POE module also has some other issue which affects its networking.

**Reply by magichew** [reply]
I think I have things progressing now. It has taken having only one device on the network at a time to configure. I've followed the settings that you've outlined in the read my. My question is on the TCPServer side... What should the destination IP point to and which port? Is that to point to Home Assistant?

**Reply by magichew** [reply]
Also, I've tried changing the 'Dev Name' to something that would be distinctive but it doesn't seem to be saved when I hit 'modify setting'. All other parameters seem to be saved okay.

**Reply by alienatedsec** [reply]
@magichew 
TCP Server should have TCP Client greyed out. It's a Server so it expects connections and does not initiate one. Set your Home Assistant integration to connect to its IP and Port.

**Reply by alienatedsec** [reply]
I just checked myself in the app and on the screenshots - it's not greyed out for some reason, but I would say it doesn't need any IPs

**Reply by magichew** [reply]
Yes, not greyed out on mine either. You've obscured your settings for Dest IP/Domain and Dest. Port. Would you be willing to share?  At present mine are pointing to 192.168.1.3 and 4196. I'm not using the 192.168.1.xxx range on my network so wouldn't go anywhere.

**Reply by alienatedsec** [reply]
It's not working if TCP Server is selected. I obscured those as they are not related. You may stay using the same settings as you already have.

**Reply by magichew** [reply]
Finally I have both showing in Vircom. I've set reservation in DHCP for both devices and that seems to have done the trick. Is there a way to test that they can communicate with each other before I get the soldering iron out for the socket?

**Reply by alienatedsec** [reply]
@magichew Try this https://www.waveshare.com/wiki/RS485_TO_ETH_(B)#TCP_Communication_Test

However, I would not waste time on it. If you can ping both devices - that should be more than enough unless you have some multi-level firewall system, which needs a more in-depth ruleset to be applied.

**Reply by hazbot72** [reply]
@magichew just a heads up that you might want to use static IPs instead of DHCP. 

I usually prefer DHCP myself but when I set up the Waveshares that way, I found they did not renew their DHCP lease after it expired (24 hours in my case).  Most other devices will send a DHCP renew but these apparently don't.

That caused an issue in my setup because the DNS entry for the Waveshares expired at the same time as the DHCP lease ended. I switched to static IPs to fix this.

**Reply by magichew** [reply]
Thanks for the tip. Is there a difference between setting a DHCP reservation and setting a static IP?

**Reply by alienatedsec** [reply]
@magichew The static IP is set and does not change. The IP reservation requires a device to ask for an IP. 

Mine seems to be consistent and works all the time. It may be related to a DHCP server. @hazbot72 made some points and it could be worth considering.

**Reply by hazbot72** [reply]
I guess it depends on your network setup and which device(s) provide DHCP and DNS.  In my case I use Pi-hole which acts as both. Only problem with that is as above ... Pi-hole only maintains DNS entries for each device for as long as there is also a DHCP lease for that device.

If a device allows its DHCP lease to expire and doesn't renew it, then the DNS entry pointing to that device also expires.

**Reply by magichew** [reply]
I believe I've set static IP on the Waveshare and also set DHCP reservation in my router. Should mean that they get the same IP address every time yes?

I have blue link light on each unit now. Now I need to work on the Home Assistant side of things.

**Reply by hazbot72** [reply]
Yes, setting static IP on the Waveshares themselves means they won't even request a DHCP address and just use the given IP all the time.

But it is good to also have the fixed DHCP reservation on your router, because it means no other device could be accidentally given the same IP as either of the Waveshares.
